packages/debug-ui/scripts/tcrActions.ts

```diff
@@ -1,33 +1,138 @@
 import { Civil } from "@joincivil/core";
+import { getVoteSaltHash } from "@joincivil/utils";
 import BigNumber from "bignumber.js";
-import { EthAddress } from "../../core/build/src/types";
+import { Bytes32, EthAddress } from "../../core/build/src/types";
 
 export async function apply(address: EthAddress, optionalCivil?: Civil): Promise<void> {
   const civil = optionalCivil || new Civil();
 
   console.log("Apply to TCR");
   const tcr = await civil.getDeployedOwnedAddressTCRWithAppeals();
 
-  const eip20Address = await tcr.getTokenAddress();
-  const eip = await civil.eip20AtUntrusted(eip20Address);
+  const eip = await civil.getEIP20ForDeployedTCR();
 
   const approvedTokensForSpender = await eip.getApprovedTokensForSpender(tcr.address);
+  console.log("approvedTokensForSpender: " + approvedTokensForSpender);
   if (approvedTokensForSpender.toNumber() < 1000) {
-    await eip.approveSpender(tcr.address, new BigNumber(1000 - approvedTokensForSpender.toNumber()));
+    const tokensToApprove = 1000;
+    console.log("approve this many tokens: " + tokensToApprove);
+    const approveTransaction = await eip.approveSpender(tcr.address, new BigNumber(tokensToApprove));
+    await approveTransaction.awaitReceipt();
   }
 
-  const { awaitReceipt } = await tcr.apply(address, new BigNumber(1000), "test");
-  const applyReceipt = await awaitReceipt;
+  const applyTransaction = await tcr.apply(address, new BigNumber(1000), "test");
+  await applyTransaction.awaitReceipt();
   console.log("Applied to TCR");
 }
 
+export async function challenge(address: EthAddress, optionalCivil?: Civil): Promise<void> {
+  const civil = optionalCivil || new Civil();
+
+  console.log("Challenging TCR Listing");
+  const tcr = await civil.getDeployedOwnedAddressTCRWithAppeals();
+
+  const eip = await civil.getEIP20ForDeployedTCR();
+
+  const approvedTokensForSpender = await eip.getApprovedTokensForSpender(tcr.address);
+  if (approvedTokensForSpender.toNumber() < 1000) {
```

Comment:
```typescript
if (approvedTokensForSpender.isLessThan("1000"))
```

Reply:
Also don't use constants here, since this script might work on a different parameters from parametizer

Reply:
The value 1000 is totally placeholder. Getting the actual value from parameterizer will be next thing I do (along with the rest of the parameterizer). I'll add a TODO here though.